PhpUnit/DefinitionHasArgumentConstraint.php

```diff
@@ -17,15 +17,15 @@ public function __construct($argumentIndex, $expectedValue, $checkExpectedValue
     {
         parent::__construct();
 
-        $this->argumentIndex = (integer)$argumentIndex;
+        $this->argumentIndex = $argumentIndex;
         $this->expectedValue = $expectedValue;
         $this->checkExpectedValue = $checkExpectedValue;
     }
 
     public function toString()
     {
         return sprintf(
-            'has an argument with index %d with the given value',
+            'has an argument with index %s with the given value',
```

Comment:
I would not talk about "index" here if `$this->argumentIndex` is a string

Reply:
"Key" would be the first thing in my mind to discribe it. Or is there a more known term in symfony?

Reply:
Why not "has no argument named %s" then?